tests/rptest/services/redpanda.py

```diff
@@ -646,3 +646,38 @@ def delete_topic(self, name):
 
     def cov_enabled(self):
         return self._context.globals.get(self.COV_KEY, self.DEFAULT_COV_OPT)
+
+    def transfer_partition_leadership(self,
+                                      topic,
+                                      partition,
+                                      target,
+                                      timeout_sec=30):
```

Comment:
I'd suggest not having a default timeout here -- make any caller explicitly state their expectation, as it is a success condition of the test.